File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int n = s.size();
        int i = n - 1;

        // Step 1: Skip trailing spaces
        while (i >= 0 && s[i] == ' ') {
            i--;
        }

        // Step 2: Count characters of the last word
        int length = 0;
        while (i >= 0 && s[i] != ' ') {
            length++;
            i--;
        }

        return length; // Step 3: Return the count
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        i = len(s) - 1

        # Step 1: Skip trailing spaces
        while i >= 0 and s[i] == ' ':
            i -= 1

        # Step 2: Count characters of the last word
        length = 0
        while i >= 0 and s[i] != ' ':
            length += 1
            i -= 1

        return length  # Step 3: Return the count
